Sources/NIOSSHServer/main.swift

```diff
@@ -62,18 +62,26 @@ defer {
 func sshChildChannelInitializer(_ channel: Channel, _ channelType: SSHChannelType) -> EventLoopFuture<Void> {
     switch channelType {
     case .session:
-        return channel.pipeline.addHandler(ExampleExecHandler())
+        return channel.eventLoop.makeCompletedFuture {
+            try channel.pipeline.syncOperations.addHandler(ExampleExecHandler())
+        }
     case .directTCPIP(let target):
-        let (ours, theirs) = GlueHandler.matchedPair()
+        return channel.eventLoop.makeCompletedFuture {
+            let (ours, theirs) = GlueHandler.matchedPair()
+            _ = channel.pipeline.addHandler(DataToBufferCodec())
+            try channel.pipeline.syncOperations.addHandler(ours)
+
+            let loopBoundHandler = NIOLoopBound(theirs, eventLoop: channel.eventLoop)
 
-        return channel.pipeline.addHandlers([DataToBufferCodec(), ours]).flatMap {
-            createOutboundConnection(
+            _ = createOutboundConnection(
```

Comment:
We're ignoring the future here so the future returned from `sshChildChannelInitializer` can complete before all of the intended work is done. 

You'll need to chain this off of the `channel.eventLoop.makeCompletedFuture { ... }`